runtime/src/snapshot_utils.rs

```diff
@@ -2882,6 +2882,7 @@ pub fn verify_snapshot_archive<P, Q, R>(
 pub fn purge_old_bank_snapshots(
     bank_snapshots_dir: impl AsRef<Path>,
     num_bank_snapshots_to_retain: usize,
+    type_select: Option<BankSnapshotType>,
```

Comment:
Sure, I can make the change.  But I also think the current way should be fine.  type_select means there is a specify type to select.  if it is None, that means when purging there is no specific type to select, so it applies to all the types.

Here is the answer from ChatGPT
One way to express a parameter that can either select a specific type or apply to all types in Rust is to use the Option type.

You can define your function to take an Option<T> parameter, where T is the type you want to apply the operation to. If the parameter is Some(value), you apply the operation to that value. If the parameter is None, you apply the operation to all values of type T.

Here's an example implementation:
```
fn apply_operation<T: std::ops::Add<Output = T>>(vector: &mut Vec<T>, value: Option<T>) {
    match value {
        Some(val) => {
            for element in vector.iter_mut() {
                *element = *element + val;
            }
        }
        None => {
            let default_val = T::default();
            for element in vector.iter_mut() {
                *element = *element + default_val;
            }
        }
    }
}

Reply:
> Sure, we can do three states with Option, but it also reads strangely to me that None means all/both.

Agree with this - although it might just be a naming thing for me? `type_select`, to me, implies passing in the type(s) I want to purge, and `None` doesn't mesh with that mental model well.

> purge_old_bank_snapshots_pre(), which only purges PRE
purge_old_bank_snapshots_post(), which only purges POST
purge_old_bank_snapshots(), which purges both (and the impl just calls the two above)?

I personally think this is the best option - `do_purge` will need to become a function as well.

Reply:
To follow this suggestion, then do_purge should be factored out to avoid duplication.

Then, to make do_purge a function, the function name reflecting its content should be purge_old_bank_snapshots, same as the existing one.  The only difference is that the existing one takes snapshots_dir as the input parameter, not a vector of BankSnapshotInfo's.  This looks quite confusing.

Maybe we should just replace the existing purge_old_bank_snapshots(bank_snapshots_dir, ..) with purge_old_bank_snapshots(bank_snapshots: Vec<BankSnapshotInfo>, ..), which has the do_purge body?

Reply:
> > Sure, we can do three states with Option, but it also reads strangely to me that None means all/both.
> 
> Agree with this - although it might just be a naming thing for me? `type_select`, to me, implies passing in the type(s) I want to purge, and `None` doesn't mesh with that mental model well.
>

If "type_select" is replaced with "filter_by_type", will it be easier to understand?  It can be read as:
if filter_by_type is None, there is no filtering;
if filter_by_type is Some(type), then filter by that type.

Reply:
I'm content with that naming. @brooksprumo?

Reply:
> I'm content with that naming. @brooksprumo?

Option 1 is about function naming.  Option 2 is about the parameter naming.  Not sure which one you are referring to.  :)

Reply:
The filter_by_type approach

Reply:
updated with filter_by_type